### PR TITLE
refactor: use `options` as second parameter

### PR DESCRIPTION
**Pull request by lwhiteley**
- add support options object to configure `proxyWithHistory`
- extract shouldSaveHistory utility

### CODE REVIEW

**Comment by dai-shi** [comment]
You can keep supporting boolean value and show deprecation warning.

**Comment by lwhiteley** [comment]
updated

**Comment by lwhiteley** [comment]
in the end, i re-added everything except the unsubscribe handler.

Should have enough convenience features now.

```ts
proxyWithHistory(data, { subscribeCallback: console.log })
```

File: packages/history-utility/src/history-utility.ts [hunk 1]
```diff
@@ -38,6 +39,24 @@ export type History<T> = {
...
+export type HistoryOptions = {
+  /**
+   * determines if the internal subscribe behaviour should be skipped.
+   */
+  skipSubscribe?: boolean;
+
+  /**
+   * a callback function executed on each subscribe call
+   */
+  subscribeCallback?: HistorySubscribeCallback;
```
Comment (lwhiteley) [inline packages/history-utility/src/history-utility.ts:57]: should we call this `onHistorySaved` and only call it when the history is stored?

cc: @dai-shi

Comment (lwhiteley) [reply packages/history-utility/src/history-utility.ts]: scrap that..actually want the callback to be triggered everytime

File: packages/history-utility/src/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const isProduction = import.meta?.env?.MODE === 'production';
```
Comment (dai-shi) [inline packages/history-utility/src/helpers.ts:1]: This doesn't work well for minification. We should simply have inline `if (...)`.

Comment (lwhiteley) [reply packages/history-utility/src/helpers.ts]: updated

**Review by dai-shi — APPROVED** [review]
just reviewed `packages/history-utility/src/history-utility.ts`.

**Pull request merged**